wp-cache-sanitizer.php

```diff
@@ -0,0 +1,726 @@
+                $sanitized_value = crc32( $WPSC_HTTP_HOST . $cache_path ) & 0x7fffffff;
+            }
+            break;
+
+        default:
+            $sanitized_value = apply_filters( "wp_super_cache_sanitize_$value", $value );
+            $sanitized_value = intval( $sanitized_value );
+            break;
+    }
+    return $sanitized_value;
+  }
+
+  /**
+   * This sanitizes the settings where the allowed values are only strings.
+   * The sanitation method is different per setting name.
+   *
+   * @param String $setting The current string setting to be sanitized
+   * @param Mixed $value The mixed value of the current setting that will be sanitized
+   *
+   * @uses  wp_cache_sanitize_value Sanitizes an array value
+   *
+   * @return String|Array A sanitized version of the inputed value
+   */
+  function sanitize_string( $setting, $value ) {
+    switch ( $setting ) {
+
+        case 'cache_acceptable_files' :
+            global $cache_acceptable_files;
+            if ( is_array( $value ) ) {
+                $value = implode( ',', $value );
+            }
+            $sanitized_value = wp_cache_sanitize_value( $value, $cache_acceptable_files );
+            break;
+
+        case 'cache_badbehaviour_file':
+            $sanitized_value = function_exists('get_bb_file_loc') ? get_bb_file_loc() : "";
+            $sanitized_value = "\"$sanitized_value\"";
+            break;
+
+        case 'cache_no_adverts_for_friends':
+            $sanitized_value = in_array( $value, array( 'yes', 'no' ) ) ? $value : "yes";
+            $sanitized_value = "\"$sanitized_value\"";
+            break;
+
+        case 'cache_rejected_user_agent':
+            global $cache_rejected_user_agent;
+            if ( is_array( $value ) ) {
+                $value = implode( ',', $value );
+            }
+            $value = str_replace( ' ', '___', $value );
+            $sanitized_value = str_replace( '___', ' ', wp_cache_sanitize_value( $value, $cache_rejected_user_agent ) );
+            break;
+
+        case "cache_scheduled_time":
+            $sanitized_value = preg_match( '/^\d{2}:\d{2}$/', $value ) ? "\"$value\"" : "\"00:00\"";
+            break;
+
+        case "cache_schedule_interval":
+            $sanitized_value = in_array( $value, array( 'daily', 'twicedaily', 'hourly' ) ) ? "\"$value\"" : "\"daily\"";
+            break;
+
+        case 'cache_schedule_type':
+            $sanitized_value = in_array( $value, array( 'time', 'interval' ) ) ? "\"$value\"" : "\"interval\"";
+            break;
+
+        case 'cache_page_secret':
+            if ( strlen( $value ) != 32 || ! ctype_xdigit( $value ) ) {
+                $sanitized_value = md5( date( 'H:i:s' ) . mt_rand() );
+            } else {
+                $sanitized_value = $value;
+            }
+            $sanitized_value = "\"$sanitized_value\"";
+            break;
+
+        case 'cached_direct_pages':
+            if ( is_array( $value ) )  {
```

Comment:
The settings that are arrays are outliers in the boolean, int, string paradigm. Since the arrays contain strings in them I decided to group them with the string settings. This is also the case for the other array settings and the `$wp_cache_pages` array as well. There is a case for regrouping these array settings separately.